feat(pipeline-ui): Display spark graph in destinations table

tiina303 opened the pull request:
## Problem

<!-- Who are we building for, what are their needs, why is this important? -->
We only showed 24h before, but that's not super informative when people don't use PostHog daily
<img width="216" alt="Screenshot 2024-01-11 at 15 57 56" src="https://github.com/PostHog/posthog/assets/890921/5ee7481e-de1a-48c5-abfb-5ea22de5c380">



## Changes

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

tiina303 committed:
feat(pipeline-ui): Display spark graph in destinations table

thmsobrmlr reviewed (dismissed):
Love the spark line idea & would like to reuse it on the overview page. Would slice the frontend components a bit different though (see comment).

thmsobrmlr commented on frontend/src/lib/lemon-ui/LemonTable/TableCellSparkline.tsx:
```diff
@@ -4,9 +4,28 @@ import { offset } from '@floating-ui/react'
+    values: number[]
+}
+
+function isSparkLineTimeSeries(data: number[] | SparkLineTimeSeries[]): data is SparkLineTimeSeries[] {
```
I think it'd be better for long term maintainability to extract a generic `Sparkline` / `LemonSparkline` component instead of adapting this inner component and re-using it in a different scenario.

Twixes replied:
I don't think there's anything table-specific in this component, so pulled it out into `lib/lemon-ui/` as simply `Sparkline`. Could use a bit of visual tightening but should be useful generally.

Twixes committed:
Merge branch 'master' into pipeline-ui-spark-graph

Twixes committed:
Sync lint-staged paths with CI

Otherwise couldn't even merge with master due to `bin/build-schema.mjs` being linted in the pre-commit hook despite not being included by `tsconfig.json`. CI didn't notice this because it never ran on that file - so we shouldn't run on it locally either.

Twixes committed:
Fix `pipelineAppMetricsLogic`

Twixes committed:
Generalize `Sparkline`

thmsobrmlr committed:
run prettier

Twixes committed:
Mock metrics in all stories

Twixes committed:
Speed Storybook up slightly

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Twixes committed:
Merge branch 'master' into pipeline-ui-spark-graph
